webhooks/mutatepod_parts.go

```diff
@@ -77,7 +83,8 @@ func gcloudSetupContainer(
 		}, {
 			Name:  "CLOUDSDK_CONFIG",
 			Value: GCloudConifgMountPath,
-		}},
+		}, projectEnvVar(project)},
+		SecurityContext: securityContext,
```

Comment:
I think `SecurityContext: nil` and `SecurityContext: &corev1.SecurityContext{}` are different.

`SecurityContext: &corev1.SecurityContext{}`: empty securitycontext has default int values. So, this runs with `uid=0`
`SecurityContext: nil`: security context is just not set. So, the default user/group of specified container image will be respected

Reply:
I have updated the code, if `RunAsUser == nil` now we set `SecurityContext: nil`